tests/byzantium/eip198_modexp_precompile/helpers.py

```diff
@@ -86,6 +128,26 @@ def from_bytes(cls, input_data: Bytes | str) -> "ModExpInput":
 
         return cls(base=base, exponent=exponent, modulus=modulus, raw_input=input_data)
 
+    @classmethod
+    def create_mismatch(
```

Comment:
This feels a bit too much for me. I generally struggle with some data types in EEST which have a lot of interconnected options and flags.

This could be a normal constructor. And you can fix the properties directly in the constructor like:

```python
length_base = declared_base_length if  declared_base_length not None else len(base)
```